source-dropbox/oauth2.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/dropbox/dropbox-sdk-go-unofficial/v6/dropbox"
+	"golang.org/x/oauth2"
+)
+
+// TODO: Replace the client ID, client secret and redirect URL with the real ones
+var (
+	dropboxConfig = &oauth2.Config{
+		Scopes: []string{
+			"files.metadata.read",
+			"files.content.read",
+			// "account_info.read",  // Not sure if we need this
+		},
+		Endpoint:     dropbox.OAuthEndpoint(""),
+		ClientID:     "client ID goes here",
+		ClientSecret: "client secret goes here",
+		RedirectURL:  "http://localhost:8080/oauth2/callback",
+	}
+)
+
+func GetOAuth2HTTPClient(ctx context.Context, config *oauth2.Config) (*http.Client, error) {
+
+	// token := new(oauth2.Token)
+	// tokenSource := config.TokenSource(ctx, token)
+
+	// use PKCE to protect against CSRF attacks
+	// https://www.ietf.org/archive/id/draft-ietf-oauth-security-topics-22.html#name-countermeasures-6
+	verifier := oauth2.GenerateVerifier()
+
+	// TODO: Handle the code in the UI
```

Comment:
I think most of what is described as being done here is automatically done via the UI already. Again, checkout out `source-google-drive` etc. and see how that works. The connector's responsibility is to produce a valid oauth spec, but it shouldn't need to actually navigate the oauth flow.